internal/cli/finalizer.go

```diff
@@ -0,0 +1,54 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+const timeout = 5
+
+type Finalizer struct {
+	notify func(c chan<- os.Signal, sig ...os.Signal)
+	exit   func(int)
+	funcs  []func()
+}
+
+func NewFinalizer() *Finalizer {
+	return &Finalizer{
+		notify: signal.Notify,
+		exit:   os.Exit,
```

Comment:
This field is not needed, you can just call `os.Exit()` directly in the relevant place.

Reply:
This function included in the struct body because testing purposes. I'm replacing this method with my fixed method during tests.